examples/src/main/java/io/confluent/rest/examples/hellooidc/HelloOpenID.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest.examples.hellooidc;
+
+import java.util.TreeMap;
+import javax.ws.rs.core.Configurable;
+
+import org.eclipse.jetty.util.resource.Resource;
+import org.eclipse.jetty.util.resource.ResourceCollection;
+import org.glassfish.jersey.servlet.ServletProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.confluent.rest.Application;
+import io.confluent.rest.auth.OpenIdConnectConfig;
+import io.confluent.rest.examples.helloworld.HelloWorldResource;
+import io.confluent.rest.examples.helloworld.HelloWorldRestConfig;
+import io.confluent.rest.RestConfig;
+import io.confluent.rest.RestConfigException;
+
+/**
+ * An application represents the configured, running, REST service. You have to provide two things:
+ * a configuration (to the constructor or by overriding configure()) and a set of resources to
+ * for the REST API (added in setupResources()). After defining these, simply call
+ * Application.createServer() to get a Jetty server, then call start() to start processing requests
+ *
+ * <p>This application uses a simple configuration that allows you to override the message that is
+ * echoed back in the response, and the driver program optionally loads this setting from a command
+ * line argument.
+ */
+public class HelloOpenID extends Application<HelloWorldRestConfig> {
+  private static Logger log = LoggerFactory.getLogger(HelloOpenID.class);
+  private static TreeMap<String,String> settings = new TreeMap<String,String>();
+
+  public HelloOpenID(HelloWorldRestConfig config) {
+    super(config);
+  }
+
+  @Override
+  public void setupResources(Configurable<?> config, HelloWorldRestConfig appConfig) {
+    config.register(new HelloWorldResource(appConfig));
+    config.property(ServletProperties.FILTER_STATIC_CONTENT_REGEX, "/(static/.*|.*\\.html|)");
+  }
+
+  @Override
+  protected ResourceCollection getStaticResources() {
+    return new ResourceCollection(Resource.newClassPathResource("static"));
+  }
+
+  public static void main(String[] args) {
```

Comment:
So looks like this was manually tested?

I.e. you can use this example  call the main method and pass params.

Can you comment in this file or in the commit message
* An example of how you ran this, (against Okta)?
* What the expected behavior is if no creds are are passed, if the creds are wrong, if the creds are right.

Reply:
Yea I wasn't super sure how to write a test for this. I'll see if there is a mock OIDC provider in the jetty code base somewhere. If memory serves they use google as a provider in their tests which isn't super awesome. I'll try to think of something 

1. OIDC (the protocol) isn't vendor specific, but their admin apis are. We could link their docs I suppose. In the long run(not the initial PR) I'd like to support dynamic client registration in rest utils but thats for another day. 

2. Ack will add some comments around this.

Reply:
found a mock OpenIdProvider I think 👍